### PR TITLE
Script to warn when viewing version other than stable

### PR DESCRIPTION
**Pull request by jnothman**
This script would be available to work with the following change to all HTML files here:
```
$ find */ -name "*.html" | xargs sed -i '/<\/body>/ i \
\    <script src="https://scikit-learn.org/versionwarning.js"></script>'
```

It produces banners like the following on the dev version:

image

and the following on older versions:

image

Inserting the script would be done in another PR, and as a change in our Circle CI doc deployment.

To do after merge:
* [ ] Add script to *.html in released directories.
* [ ] Add script during doc build/deploy for all non-pr builds (https://github.com/scikit-learn/scikit-learn/pull/14634)
* [ ] Amend maintainer docs to make sure the latest release variable is updated with minor releases (https://github.com/scikit-learn/scikit-learn/pull/14634)
* [ ] Road test and iterate on the script.

### CODE REVIEW

**Comment by jnothman** [comment]
I think the wording and styling could benefit from some improvements. But because it's not embedded in the HTML, some of that can be improved over time.

File: versionwarning.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            ].join('; ') +
+            '">' + msg + '</p>')
+    };
+    if (location.hostname == 'scikit-learn.org') {
+        var versionPath = location.pathname.split('/')[1];
+        if (!goodPaths.includes(versionPath)) {
+            showWarning('You are looking at documentation for an old release of ' +
+                        'Scikit-learn (version ' + versionPath + '). Try the ' +
+                        '<a href="https://scikit-learn.org">latest release</a> or ' +
+                        '<a href="https://scikit-learn.org/dev">development</a> ' +
```
Comment (rth) [inline versionwarning.js]: I am not sure for putting the dev on outdated versions. Few users need it and it can lead to confusion.

**Comment by rth** [comment]
Do you want to run the sed command in this PR or a separate one?

File: versionwarning.js (hunk 1 above)
Comment (adrinjalali) [reply versionwarning.js]: hmm, the dev version would be nice if we also tell people that they can have it through the nightly wheels. We would also get more testers before the release that way. I wouldn't mind that.

**Comment by jnothman** [comment]
I tried including the sed in this branch but my web browser was a bit
overwhelmed by the diff. I think we'll do that without a pr :D

In fact, I have another branch open with that change.

I think there are several improvements that can be made here. For example a
JavaScript fetch could confirm whether an equivalent page exists in stable
and link directly to that. We could have a more complete version switcher.

I think here we mostly need to decide if this is an acceptable first
iteration. Does the dev warning look too much like the old version warning?

File: versionwarning.js (hunk 1 above)
Comment (jnothman) [reply versionwarning.js]: Do you mean that the warning shown on dev should point the user to installation instructions? Nice idea

Comment (NicolasHug) [reply versionwarning.js]: I agree with @rth about not strictly encouraging the dev docs. Maybe we can make it clearer with

> Try the latest stable release (recommended) or the development version (unstable).

And yes linking to installation instructions on the dev version would be great too

**Comment by jnothman** [comment]
Updates:
image

image

**Comment by adrinjalali** [comment]
Nice, I like this one, shall we merge?

**Comment by NicolasHug** [comment]
> shall we merge?

+1, I haven't reviewed the JS but I like the messages too

**Comment by jnothman** [comment]
Now live on non-dev versions. https://scikit-learn.org/0.20/

File: versionwarning.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    var goodPaths = ['stable', 'dev', latestStable];
+    var showWarning = (msg) => {
+        $('.body[role=main]').prepend(
+            '<p style="' +
+            [
+                'padding: 1em',
+                'font-size: 1.5em',
+                'border: 1px solid red',
+                'background: pink',
+                'width: 100%',
```
Comment (GaelVaroquaux) [inline versionwarning.js]: This specific line makes the box non mobile-phone friendly (it creates a horizontal scroll by extending out of the viewport). It is not needed.

Comment (GaelVaroquaux) [reply versionwarning.js]: I opened the following PR to address this problem: https://github.com/scikit-learn/scikit-learn.github.io/pull/14

**Pull request merged**